cinder/volume/drivers/lightos.py

```diff
@@ -1062,23 +1075,36 @@ def __add_volume_acl(self, project_name, lightos_volname, acl_to_add):
             LOG.warning('Got LightOS volume without ACL?! data: %s', data)
             return False
 
+        ip_acl= data.get('IPAcl')         
+        if not ip_acl:
+           LOG.warning('Got LightOS volume without IP ACL?! data: %s', data)
+           return False
+
         acl = acl.get('values', [])
+        ip_acl = ip_acl.get('values', [])
+        
 
         # remove ALLOW_NONE and add our acl_to_add if not already there
         if 'ALLOW_NONE' in acl:
             acl.remove('ALLOW_NONE')
         if acl_to_add not in acl:
-            acl.append(acl_to_add)
+           acl.append(acl_to_add)
 
+        ip_acl=[]    
+        if self.configuration.lightos_use_ipacl:
```

Comment:
@muliby-lb  
Every volume is created with initial IP-ACL value "ALLOW_NONE",  and when volume is connected then this check will ensure if the flag **lightos_use_ipacl**  in cinder.conf is set to True or False, if this variable lightos_use_ipacl=true then code will write IPs to IP-ACL otherwise will set volume IP-ACL as ALLOW_ANY.

Reply:
I'm not sure this is correct -- we are in *add*, not *overwrite*, so the IPACL should be added if use_ipacl is True and left untouched if use_ipacl is false.

Reply:
@muliby-lb 
what would you recommend the volume's IP-ACL initial value when 
 -  lightos_use_ipacl=True   and  volume is created, but has not attached yet.
-   lightos_use_ipacl=False  and  volume is created, but has not attached yet.

Prior to this change ip-acl has alway been created with initial value ALLOW_ANY.

Reply:
`lightos_use_ipacl = True` => initial IPACL is `ALLOW_NONE` or equivalent
`lightos_use_ipacl = False` => initial IPACL is `ALLOW_ANY`